core/process.py

```diff
@@ -1499,7 +1519,13 @@ def WhenDone(self, pid):
         if pid == self.last_stopped_pid:
             self.last_stopped_pid = -1
 
-        mylib.dict_erase(self.jobs, pid)
+        assert pid in self.child_procs
+        proc = self.child_procs[pid]
+        if proc.job_id in self.jobs:
+            mylib.dict_erase(self.jobs, proc.job_id)
```

Comment:
the dict_erase() can go outside the `if`

The API of dict_erase() is that it ensures the key isn't present unconditionally